worker/modelupgrader/worker.go

```diff
@@ -174,8 +189,14 @@ func newUpgradeWorker(config Config, targetVersion int) (worker.Worker, error) {
 			targetVersion,
 			setVersion,
 		); err != nil {
+			if err := setStatus(status.Error, "failed to upgrade environ"); err != nil {
```

Comment:
I thought it might be TMI for non-controller admins. I'll include it, we can change based on feedback.